src/power.rs

```diff
@@ -56,6 +137,27 @@ impl Power {
     }
 }
 
+macro_rules! wakeup_pins {
+    ($($PIN:path: $line:expr,)+) => {
+        $(
+            impl<M> From<&$PIN> for WakeUp {
+                fn from(_: &$PIN) -> Self {
+                    $line
+                 }
+            }
+        )+
+    }
+}
+
+wakeup_pins! {
+    gpioa::PA0<M>: WakeUp::Line1,
```

Comment:
I'd move `<M>` to the macro above

Reply:
I can't do that, cause macro require full type :-(

Reply:
Reexport all `PXX` types from `gpio::gpiox` to `gpio`. This simplifies a lot of things

Reply:
https://github.com/stm32-rs/stm32f4xx-hal/blob/12b5bc13ab0b5ce96f01cb239e76aecf64d396d7/src/gpio.rs#L522